src/jobs/actions/urlaction.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  Logger,
+  NotFoundException,
+  BadRequestException,
+  HttpException,
+} from "@nestjs/common";
+import { JobAction } from "../config/jobconfig";
+import { JobClass } from "../schemas/job.schema";
+import * as Handlebars from "handlebars";
+
+// Handlebar options for JobClass templates
+// TODO should this be moved into job.schema.ts?
+const jobTemplateOptions = {
+  allowedProtoProperties: {
+    id: true,
+    type: true,
+    statusCode: true,
+    statusMessage: true,
+    messageSent: true,
+    createdBy: true,
+    jobParams: false,
+  },
+  allowProtoPropertiesByDefault: false, // limit accessible fields for security
+};
+
+/**
+ * Respond to Job events by making an HTTP call.
+ */
+export class URLAction<T> implements JobAction<T> {
+  public static readonly actionType = "url";
+
+  private urlTemplate: Handlebars.TemplateDelegate<JobClass>;
+  private method = "GET";
+  private headers: Record<string, string> = {};
+  private body: Record<string, any> | null = null;
```

Comment:
Body probably also needs to accept templates, but how to deal with it in JSON input? Maybe only apply templating to string values?